### PR TITLE
Refactor addToCart and goToCheckout actions

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Trying to reduce complexity of adding offer to cart
- Remove form and local "redirect target" state, make submit a simple `onClick` handler instead
- Wait until transition to cart completes before cleaning up old price intent
- Less prop-drilling and indirection

Video below shows how "go to checkout" looks after these changes. Note that loading state sticks right until checkout page opens

<div class='graphite__hidden'>
          <div>🎥 Video uploaded on Graphite:</div>
            <a href="https://app.graphite.dev/media/video/GLY3tSM85RFBvAf6lFNA/515b7d84-ac77-4bca-adb4-929be3f12c97.mov">
              
            </a>
          </div>
<video src="https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/GLY3tSM85RFBvAf6lFNA/515b7d84-ac77-4bca-adb4-929be3f12c97.mov">Screen Recording 2024-05-31 at 11.39.01.mov</video>

## Justify why they are needed

This hopefully fixes problem with add to cart instability that we've seen during app router test session

Whole state management of purchase form remains horrible, unstable and very hard to understand. We should refactor it some time in the future

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

File: apps/store/src/components/ProductPage/PriceIntentContext.tsx [hunk 1]
```diff
@@ -81,14 +81,13 @@ const usePriceIntentContextValue = () => {
...
-  const createNewPriceIntent = useCallback(
-    async (shopSession: ShopSession) => {
-      priceIntentServiceInitClientSide(apolloClient).clear(priceTemplate.name, shopSession.id)
-      await updatePriceIntent(shopSession)
-      setSelectedOffer(null)
-    },
-    [apolloClient, priceTemplate.name, updatePriceIntent, setSelectedOffer],
-  )
+  const shopSessionId = shopSession?.id
+  const resetPriceIntent = useCallback(() => {
```
Comment (alebedev) [inline apps/store/src/components/ProductPage/PriceIntentContext.tsx:85]: We don't actually need to create new priceIntent here, it'll happen automatically based on existing effects

**Review by gustaveen — APPROVED** [review]
Feels cleaner! And very nice that the we keep the loading state now 👍 
I agree, the logic in purchase form is a bit hard to follow 🙃

**Comment by alebedev** [comment]
### Merge activity

* **May 31, 6:58 AM EDT**: @alebedev started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4325).
